server/modules/record/repository/validationReportRepository.js

```diff
@@ -1,74 +1,93 @@
-import * as R from 'ramda'
-import * as camelize from 'camelize'
-
-import * as SchemaRdb from '@common/surveyRdb/schemaRdb'
+import * as A from '@core/arena'
 import * as RecordValidation from '@core/record/recordValidation'
-
 import * as Validation from '@core/validation/validation'
 
 import { db } from '@server/db/db'
 import { getSurveyDBSchema } from '@server/modules/survey/repository/surveySchemaRepositoryUtils'
+import * as NodeKeysRepository from './nodeKeysRepository'
 
 const { prefixValidationFieldChildrenCount: prefixChildrenCount } = RecordValidation
 
 // ============== READ
 
-const query = ({ surveyId, recordUuid }) =>
-  `SELECT
-      r.uuid as record_uuid,
-      r.step as record_step,
-      r.owner_uuid as record_owner_uuid,
+const query = ({ surveyId, recordUuid }) => {
+  const surveySchema = getSurveyDBSchema(surveyId)
+  const uuidLength = 36
 
-      h.node_uuid,
-      
+  return `WITH node_validation AS (
+    SELECT 
+      r.uuid AS record_uuid,
+      -- node_uuid
+      -- if the length of the key is ${uuidLength}, then it's a uuid
+      -- otherwise the key of the field validation starts with '${prefixChildrenCount}' followed by the child def uuid
+      (
+        CASE WHEN LENGTH(nv.key) = ${uuidLength}
+        THEN nv.key
+        ELSE SUBSTRING(nv.key, ${prefixChildrenCount.length + 1}, ${uuidLength})
+        END
+      )::uuid AS node_uuid,
       -- validation_count_child_def_uuid
-      CASE WHEN LENGTH(node_validation.key) > 36 
-        THEN SUBSTRING(node_validation.key, ${prefixChildrenCount.length + 38}, 36)
+      (
+        CASE WHEN LENGTH(nv.key) > ${uuidLength}
+        THEN SUBSTRING(nv.key, ${prefixChildrenCount.length + uuidLength + 2}, ${uuidLength})
         ELSE NULL
-        END 
-        AS validation_count_child_def_uuid,
-      
-      node_validation.value::jsonb AS validation,
-
-      h.node_def_uuid,
-      h.keys_self,
-      h.keys_hierarchy
+        END
+      )::uuid AS validation_count_child_def_uuid,
+      -- node validation object
+      nv.value::jsonb AS validation
     FROM
-      ${getSurveyDBSchema(surveyId)}.record r,
-      jsonb_each(r.validation #> '{${Validation.keys.fields}}' ) node_validation
+      survey_1893.record r,
```

Comment:
be careful!
${getSurveyDBSchema(surveyId)}.record r,

Reply:
uuuuu thanks!!!!